### PR TITLE
🔨 Resolve prices.tf 401 issue

### PR DESCRIPTION
**Pull request by idinium96**
Might fix the issue where the bot doesn't even bother to get new JWT from prices.tf. Quite ugly tbh.

image

### CODE REVIEW

**Comment by RobotoLev** [comment]
Reviewing it rn

**Comment by idinium96** [comment]
Hhmmmm something is wrong. It does not wait until one job is completed.

image

image

image

File: src/lib/pricer/pricestf/prices-tf-api.ts [hunk 1]
```diff
@@ -53,24 +53,46 @@ export default class PricesTfApi {
         data?: Record<string, any>,
         headers?: Record<string, unknown>
     ): Promise<B> {
-        try {
+        return new Promise((resolve, reject) => {
             if (this.token === '') {
-                await this.setupToken();
+                this.setupToken()
+                    .then(() => {
+                        return PricesTfApi.apiRequest<B>(httpMethod, path, params, data, {
```
Comment (idinium96) [inline src/lib/pricer/pricestf/prices-tf-api.ts:60]: Probably because of this... Let me try remove it

**Comment by RobotoLev** [comment]
I had that situtation today, so I'm watching the log in order to make something more clear. I don't think that happens because of using try-catch instead of promise's .catch() thing.

**Comment by idinium96** [comment]
> I had that situtation today, so I'm watching the log in order to make something more clear. I don't think that happens because of using try-catch instead of promise's .catch() thing.

It's okay I've found the problem. But now another problem came.

**Comment by idinium96** [comment]
Looks like this doesn't help either. It doesn't even update my dummy token.
image

**Comment by RobotoLev** [comment]
So as I thought, it just doesn't re-request token for some reason. I'm pretty sure all was fine when I was implementing fix for v4.13.1.

As I see in my Discord channel history, this strange 401 wasn't appearing on v4.13.1-v5.0.0, but appeared again at v5.0.1. I still don't know what might be reason for it, but that's the pure fact.

**Comment by idinium96** [comment]
> So as I thought, it just doesn't re-request token for some reason. I'm pretty sure all was fine when I was implementing fix for v4.13.1.
> 
> As I see in my Discord channel history, this strange 401 wasn't appearing on v4.13.1-v5.0.0, but appeared again at v5.0.1. I still don't know what might be reason for it, but that's the pure fact.

For me, it started appearing on v4.11.0.
I think I am going to revert changes made in https://github.com/TF2Autobot/tf2autobot/pull/1109/commits/3cbf725086f248611e007ece37aa52e59d5f7a98

**Comment by RobotoLev** [comment]
Wait a second. Can `filterAxiosError` change error format???

**Comment by RobotoLev** [comment]
You've probably cut off `response` from `AxiosError`, so catch doesn't re-request token.

**Comment by idinium96** [comment]
> Wait a second. Can `filterAxiosError` change error format???

Ah dang, you're right. That might be the issue. 🤐

**Comment by RobotoLev** [comment]
That's definitely it. There's no `err.response.status` now (there's no `err.response` even), so bot thinks that it's non-standart error and drops it farther.

**Comment by idinium96** [comment]
> That's definitely it. There's no `err.response.status` now (there's no `err.response` even), so bot thinks that it's non-standart error and drops it farther.

Let me try once again, and then again before this.

**Comment by idinium96** [comment]
Yeah... gonna try the old one.
image

**Comment by idinium96** [comment]
Oh god, the old one works just fine
image

**Comment by RobotoLev** [comment]
That's me who was implementing it (check v4.13.1). At that time 401 was handled well, so now I knew the problem is not here.

**Comment by joekiller** [comment]
Yeah I'd revert. Because the function expects a promise returned but nothing is happening but a void call who knows when it'll resolve. It is likely executing out of order. If you want to keep the new form (but using callbacks blah that's what async is for) you'd need to return the promise instead of executing it as a void

**Comment by idinium96** [comment]
> That's me who was implementing it (check v4.13.1). At that time 401 was handled well, so now I knew the problem is not here.

Yeah, that was before the filterAxios thing.

**Comment by idinium96** [comment]
Thank you @RobotoLev!
Feel free to approve this PR.
image

**Review by RobotoLev — APPROVED** [review]

**Pull request merged**